src/memray/commands/run.py

```diff
@@ -51,6 +51,8 @@ def _run_tracker(
         try:
             if args.run_as_module:
                 runpy.run_module(args.script, run_name="__main__", alter_sys=True)
+            elif args.run_as_cmd:
+                exec(args.script, {"__name__": "__main__"})
```

Comment:
@pablogsal I don't really love this, but I can't think of any better option. It's a bit weird to set `__name__ = "__main__"` even when `sys.modules["__main__"]` is not being set to a module containing this code, for one thing.

That said, it's probably good enough, and I doubt most code would even notice that it's not _really_ running in the main module. Can you spot any cleaner way to do this?